app/helper/hedera/nft.go

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2022 LimeChain Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package hedera
+
+import (
+	"database/sql"
+	"github.com/limechain/hedera-eth-bridge-validator/app/domain/repository"
+	syncHelper "github.com/limechain/hedera-eth-bridge-validator/app/helper/sync"
+	"github.com/limechain/hedera-eth-bridge-validator/app/persistence/entity"
+	"github.com/limechain/hedera-eth-bridge-validator/app/persistence/entity/schedule"
+	"github.com/limechain/hedera-eth-bridge-validator/app/persistence/entity/status"
+	log "github.com/sirupsen/logrus"
+	"sync"
+)
+
+func ScheduledNftTxExecutionCallbacks(
+	transferRepository repository.Transfer,
+	scheduleRepository repository.Schedule,
+	logger *log.Entry,
+	id string,
+	hasReceiver bool,
+	statusResult *string,
+	wg *sync.WaitGroup,
+) (onExecutionSuccess func(transactionID, scheduleID string), onExecutionFail func(transactionID string)) {
+	onExecutionSuccess = func(transactionID, scheduleID string) {
+		logger.Debugf("[%s] - Updating db status to Submitted with TransactionID [%s].",
+			id,
+			transactionID)
+		err := scheduleRepository.Create(&entity.Schedule{
+			ScheduleID:    scheduleID,
+			Operation:     schedule.TRANSFER,
+			TransactionID: transactionID,
+			HasReceiver:   hasReceiver,
+			Status:        status.Submitted,
+			TransferID: sql.NullString{
+				String: id,
+				Valid:  true,
+			},
+		})
+		if err != nil {
+			logger.Errorf(
```

Comment:
I think that the status result should still be set to FAIL.

Reply:
If it is not set to FAIL, upon mined, it will try to update something that has not been created.